sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryItemExporter.java

```diff
@@ -66,57 +68,42 @@ private static ObjectMapper createObjectMapper() {
 
     private final TelemetryPipeline telemetryPipeline;
     private final TelemetryPipelineListener listener;
-    // TODO (trask) should this be all the resources?
-    private final Resource environmentResource;
 
     private final Set<CompletableResultCode> activeExportResults =
         Collections.newSetFromMap(new ConcurrentHashMap<>());
 
     // e.g. construct with diagnostic listener and local storage listener
     public TelemetryItemExporter(
-            TelemetryPipeline telemetryPipeline, TelemetryPipelineListener listener, Resource environmentResource) {
+        TelemetryPipeline telemetryPipeline, TelemetryPipelineListener listener) {
         this.telemetryPipeline = telemetryPipeline;
         this.listener = listener;
-        this.environmentResource = environmentResource;
     }
 
     public CompletableResultCode send(List<TelemetryItem> telemetryItems) {
-        List<List<TelemetryItem>> result =
-            groupTelemetryItemsByConnectionStringAndRoleName(telemetryItems);
+        Map<TelemetryItemBatchKey, List<TelemetryItem>> batches = splitIntoBatches(telemetryItems);
         List<CompletableResultCode> resultCodeList = new ArrayList<>();
-        for (List<TelemetryItem> batch : result) {
-            resultCodeList.add(
-                internalSendByConnectionStringAndRoleName(batch, batch.get(0).getConnectionString()));
+        for (Map.Entry<TelemetryItemBatchKey, List<TelemetryItem>> batch : batches.entrySet()) {
+            resultCodeList.add(internalSendByBatch(batch.getKey(), batch.getValue()));
         }
         return maybeAddToActiveExportResults(resultCodeList);
     }
 
     // visible for tests
-    List<List<TelemetryItem>> groupTelemetryItemsByConnectionStringAndRoleName(
+    Map<TelemetryItemBatchKey, List<TelemetryItem>> splitIntoBatches(
         List<TelemetryItem> telemetryItems) {
-        Map<String, List<TelemetryItem>> groupings = new HashMap<>();
-        // group TelemetryItem by connection string
+
+        Map<TelemetryItemBatchKey, List<TelemetryItem>> groupings = new HashMap<>();
         for (TelemetryItem telemetryItem : telemetryItems) {
+            TelemetryItemBatchKey telemetryItemBatchKey = new TelemetryItemBatchKey(
+                telemetryItem.getConnectionString(),
+                telemetryItem.getResource(),
+                telemetryItem.getResourceFromTags()
+            );
             groupings
-                .computeIfAbsent(telemetryItem.getConnectionString(), k -> new ArrayList<>())
+                .computeIfAbsent(telemetryItemBatchKey, k -> new ArrayList<>())
                 .add(telemetryItem);
         }
-
-        // and then group TelemetryItem by role name
```

Comment:
where is the code to replace this?

Reply:
role name is now included in `TelemetryItemBatchKey`, so inherently in the grouping by `TelemetryItemBatchKey`